### ISSUE TITLE
[@typescript-eslint/naming-convention] Allow single underscore character as parameter name

### ISSUE THREAD

**Issue by SylvainFI** [typescript-eslint/typescript-eslint#3565]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

Hi, I inherited a codebase with single or double underscore character as parameter. I can't figure out how to allow those in eslint.

**Repro**

```
export default reducer<IFoo[]>([], on => {
  on(Actions.showFoo, (state, action) => {
    return [...state, action.payload];
  });

  on(Actions.hideFoo, (_, __) => {
    return [];
  });
});

// also
export const getCountryList = (languageCode: string): IOption[] => { /*..*/ };

```

```JSON
{
  "rules": {
     "@typescript-eslint/naming-convention":  [
            "error",
            { "selector": ["variableLike", "enum"], "format": ["camelCase"] },
            "custom": {
                "regex": "(_|__|.*)",
                "match": true
              }
            },
          ],
  }
}
```

```TS
// your repro code case
```

**Expected Result**
no error

**Actual Result**
error  Parameter name `_` must match one of the following formats: camelCase  @typescript-eslint/naming-convention

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.27.0` |
| `@typescript-eslint/parser`        | `4.27.0` |
| `TypeScript`                       | `3.9.9` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `12.18.3` |

**Comment by bradzacher** [comment]
Have you had a read of the docs for the rule?

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#leadingunderscore--trailingunderscore
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#individual-selectors

**Comment by SylvainFI** [comment]
Yes I did, It's not about having underscore as leading or trailing character. But as a single and only character.

```
  on(Actions.hideFoo, (_, __) => {
    return [];
  });
```
